src/base/cub_ddl_log.c

```diff
@@ -0,0 +1,1206 @@
+      if (len < 0)
+	{
+	  goto write_error;
+	}
+
+      if (fwrite (buf, 1, len, fp) != len)
+	{
+	  goto write_error;
+	}
+
+      if (ddl_audit_handle->app_name == APP_NAME_LOADDB)
+	{
+	  cub_file_copy (ddl_audit_handle->file_name, dest_path);
+	}
+
+      if ((UINT64) ftell (fp) > prm_get_bigint_value (PRM_ID_DDL_AUDIT_LOG_SIZE))
+	{
+	  fclose (fp);
+	  cub_ddl_log_backup (ddl_audit_handle->log_filepath);
+	  fp = NULL;
+	}
+    }
+
+write_error:
+  if (fp != NULL)
+    {
+      fclose (fp);
+      fp = NULL;
+    }
+}
+
+void
+cub_ddl_log_write_tran_str (const char *fmt, ...)
+{
+  FILE *fp = NULL;
+  char msg[DDL_LOG_BUFFER_SIZE] = { 0 };
+  int len = 0;
+  struct timeval time_val;
+  va_list args;
+
+  if (ddl_audit_handle == NULL)
+    {
+      return;
+    }
+
+  if (prm_get_bool_value (PRM_ID_DDL_AUDIT_LOG) == FALSE)
+    {
+      return;
+    }
+
+  if (is_executed_ddl == FALSE || ddl_audit_handle->auto_commit_mode == TRUE)
+    {
+      return;
+    }
+
+  fp = cub_ddl_log_open (ddl_audit_handle->app_name);
+
+  if (fp != NULL)
+    {
+      va_start (args, fmt);
+      vsnprintf (ddl_audit_handle->msg, DDL_LOG_MSG, fmt, args);
+      va_end (args);
+
+      gettimeofday (&time_val, NULL);
+      cub_get_time_string (ddl_audit_handle->execute_start_time, &time_val);
+
+      if (ddl_audit_handle->app_name == APP_NAME_CAS)
+	{
+	  len = snprintf (msg, DDL_LOG_BUFFER_SIZE, "%s %s|%s|%s\n",
+			  ddl_audit_handle->execute_start_time,
+			  ddl_audit_handle->ip_addr, ddl_audit_handle->user_name, ddl_audit_handle->msg);
+	}
+      else if (ddl_audit_handle->app_name == APP_NAME_CSQL)
+	{
+	  len = snprintf (msg, DDL_LOG_BUFFER_SIZE, "%s %d|%s|%s\n",
+			  ddl_audit_handle->execute_start_time,
+			  ddl_audit_handle->pid, ddl_audit_handle->user_name, ddl_audit_handle->msg);
+	}
+      else
+	{
+	  goto write_error;
+	}
+
+      if (len >= DDL_LOG_BUFFER_SIZE)
+	{
+	  msg[DDL_LOG_BUFFER_SIZE - 2] = '\n';
+	  msg[DDL_LOG_BUFFER_SIZE - 1] = '\0';
+	  len = DDL_LOG_BUFFER_SIZE;
+	}
+
+      if (len < 0)
+	{
+	  goto write_error;
+	}
+
+      if (fwrite (msg, 1, len, fp) != len)
+	{
+	  goto write_error;
+	}
+
+      if ((UINT64) ftell (fp) > prm_get_bigint_value (PRM_ID_DDL_AUDIT_LOG_SIZE))
+	{
+	  fclose (fp);
+	  cub_ddl_log_backup (ddl_audit_handle->log_filepath);
+	  fp = NULL;
+	}
+    }
+
+write_error:
+  if (fp != NULL)
+    {
+      fclose (fp);
+      fp = NULL;
+    }
+
+  is_executed_ddl = FALSE;
+  cub_ddl_log_free (FALSE);
+  fprintf (stderr, "## %s\n", msg);
+}
+
+static int
+cub_create_log_mgs (char *msg)
+{
+  int retval = 0;
+  char result[20] = { 0 };
+  struct timeval exec_end, log_time;
+  int elapsed_sec = 0;
+  int elapsed_msec = 0;
+
+  if (ddl_audit_handle == NULL)
+    {
+      return -1;
+    }
+
+  gettimeofday (&exec_end, NULL);
+  cub_timeval_diff (&ddl_audit_handle->exec_begin_time, &exec_end, &elapsed_sec, &elapsed_msec);
+  cub_ddl_log_elapsed_time (elapsed_sec, elapsed_msec);
```

Comment:
**elapsed_msec** is an **int** type. But you defined that argument as long in the called function. The types do not match.
`static void cub_ddl_log_elapsed_time (long sec, long msec);`

Reply:
Thank you, I will change it to the #define statement.